src/projects/projects.controller.ts

```diff
@@ -1,48 +1,78 @@
+import { AuthGuard } from 'src/auth/guards/auth.guard';
+import { GetUser } from 'src/auth/decorators/user.decorator';
+import { GitRepoManagerService } from './git-repo-manager.service';
+import { UserDecoratorType } from 'src/auth/types/user-decorator.type';
+import { DeleteRepositoryDto } from './dto/delete-repository.dto';
 
 @Controller('projects')
 export class ProjectsController {
-    constructor(private projectsService: ProjectsService) {}
-
-    // GET /projects
-    // This action returns all of the authenticated user's projects
-    @AdminOnly()
-    @Get()
-    @HttpCode(501)
-    async findAll() {
-        // TODO: implement and remove the 501 decorator above
-        return '501 Not Implemented';
-    }
+  constructor(private projectsService: ProjectsService, private gitRepoManagerService: GitRepoManagerService) {}
 
-    // GET /projects/:uuid
-    // This action returns a #${id} project;
-    @Get(':uuid')
-    async findOne(@Param('uuid') id: string): Promise<SanitizedProject> {
-        return this.projectsService.findOne(id);
-    }
+  // GET /projects
+  // This action returns all of the authenticated user's projects
+  @Get()
+  async findAll(@GetUser() user: UserDecoratorType) {
+    return this.projectsService.findAll(user.id);
+  }
+
+  // GET /projects/:uuid
+  // This action returns a #${id} project;
+  @Get(':uuid')
+  async findOne(@Param('uuid') id: string, @GetUser() user: UserDecoratorType): Promise<SanitizedProject> {
+    return this.projectsService.findOne(id, user.id);
+  }
 
-    // POST /projects
-    /*
+  // POST /projects
+  /*
         This action creates a project
         and returns a NewProject Object
         with the project's id and name
     */
-    @Post()
-    @HttpCode(501)
-    async create(@Body() createProjectDto: CreateProjectDto) {
-        // TODO: implement and remove the 501 decorator
-        return '501 Not Implemented';
+  @Post()
+  @UsePipes(new ValidationPipe())
+  async createRepository(@Body() request: CreateProjectDto, @GetUser() user: UserDecoratorType): Promise<SanitizedProject> {
+    // Check if project name is not already used for this user
+    const project = await this.projectsService.findOneByName(request.name, user.id);
+    if (project) {
+      throw new BadRequestException('Project already exists');
     }
 
-    // DELETE /projects/:uuid
-    // This action deletes a #${id} project
-    @Delete(':uuid')
-    async delete(@Param('uuid') uuid: string) {
-        this.projectsService.delete(uuid);
-        return `OK`;
-    }
+    const createdProject = await this.projectsService.create(user.id, request);
+    // Create the project's repository
+    const projectPath = `${createdProject.uuid}`;
+    const repositoryRequest: RepositoryRequest = this.createRepositoryRequest(projectPath);
+
+    // Send the gRPC request to create the repository
+    await this.gitRepoManagerService.create(repositoryRequest);
+
+    return createdProject;
+  }
```

Comment:
Can you do the prisma query inside a transaction (example: do not create a project in db if we fail to create the git repo)?

The same could be done for `delete`)

Resource: [Prisma transaction api](https://www.prisma.io/docs/concepts/components/prisma-client/transactions#interactive-transactions)